### DISCUSSION TITLE
Using Firebase App Check Debug Provider with Angular

### DISCUSSION THREAD

**Discussion by jackwootton** [firebase/firebase-js-sdk#5757] [Q&A] [answered]
Hi,

I can  successfully initialize a web-app with a reCAPTCHA Enterprise provider. However, it's unclear to me how to use the Debug Provider from within an Angular application. The [Firebase documentation][2] recommends using the Debug Provider when testing from localhost. Specifically 

> In your debug build, enable debug mode by setting
> `self.FIREBASE_APPCHECK_DEBUG_TOKEN` to true before you initialize App
> Check. For example:

```typescript
self.FIREBASE_APPCHECK_DEBUG_TOKEN = true;
initializeAppCheck(app, { /* App Check options */ });
```

How can this be achieved in an Angular application? It's not clear what `self` is referencing or where this code snippet should be placed.

  [1]: https://stackoverflow.com/questions/67908846/how-to-initialise-firebase-app-check-in-angular
  [2]: https://firebase.google.com/docs/app-check/web/debug-provider#web-version-8_1

**Comment by jackwootton** [comment, marked as answer]
After a few experiments and guesses, the following worked for me.

Add the following to your `app.component.ts` constructor

```typescript
(<any>window).FIREBASE_APPCHECK_DEBUG_TOKEN = true;
```

**Reply by TBrayner** [reply]
Where should I place my Debut "Token code itself?
